pkg/odo/cli/component/devfile.go

```diff
@@ -165,9 +165,59 @@ func (do *DeployOptions) DevfileBuild() (err error) {
 }
 
 func (do *DeployOptions) DevfileDeploy() (err error) {
-	// TODO:
+	// Parse devfile
+	devObj, err := devfileParser.Parse(do.DevfilePath)
+	if err != nil {
+		return err
+	}
+
+	componentName, err := getComponentName(do.componentContext)
+	if err != nil {
+		return errors.Wrap(err, "unable to get component name")
+	}
```

Comment:
Set a diff componentname for now like `deploy-<componentName>`. The push passes uses the raw `componentName`

Reply:
OK, will change that and re-push the code.